pymc/sampling/mcmc.py

```diff
@@ -305,7 +305,11 @@ def _sample_external_nuts(
                 "`var_names` are currently ignored by the nutpie sampler",
                 UserWarning,
             )
-        compiled_model = nutpie.compile_pymc_model(model)
+        compiled_model = nutpie.compile_pymc_model(
+            model,
+            backend=nuts_sampler_kwargs.pop("backend", None),
```

Comment:
Depending on the outcome of https://github.com/pymc-devs/nutpie/pull/151 it may be:
```suggestion
            backend=nuts_sampler_kwargs.pop("backend", "numba"),
```

Reply:
As it happens, I submitted [a nitpick PR](https://github.com/pymc-devs/nutpie/pull/151) to fix this. It should be `None` because `None` is handled in the function.

Reply:
Otherwise you can do something like:
```python
compile_kwargs = {}
if "backend" in nuts_sampler_kwargs:
  compile_kwargs["backend"] = nuts_sampler_kwargs.pop("backend")
if "gradient_backend" in nuts_sampler_kwargs:
  compile_kwargs["gradient_backend"] = nuts_sampler_kwargs.pop("gradient_backend")
```
This way you don't have to guess the default of nutpie nor change anything there.

Reply:
Will do. The arg default still needs to be changed in nutpie one way or the other, as I discuss there.